docs/existing-aws-infrastructure.md

```diff
@@ -0,0 +1,94 @@
+
+* A VPC
+* One or more private subnets (subnets that do not automatically assign a public IP address to EC2 instances)
+* A public subnet in the same Availability Zone (AZ) for each private subnet (this is required for NAT gateways to function properly)
+* A NAT gateway for each private subnet, along with associated Elastic IP addresses
+* An Internet gateway for all public subnets
+* Route table associations that provide connectivity to the Internet through a NAT gateway (for private subnets) or the Internet gateway (for public subnets)
+
+You will need the following information, which can be obtained either via the AWS Management Console or using the AWS CLI:
+
+1. The ID of the VPC
+2. The IDs of the subnets in the VPC that you want Cluster API to use
+
+Note that there is no need to create an Elastic Load Balancer (ELB), security groups, or EC2 instances; Cluster API will take care of these items.
+
+## Tagging AWS Resources
+
+In order for Cluster API to properly recognize and consume existing AWS resources, these resources must be tagged with a specific set of tags. The list below provides the tags that must be present; in this list, `<cluster-name>` refers to the name of cluster as specified in the `metadata.name` field of the Cluster object (or manifest). Here are the tags that are required:
+
+    sigs.k8s.io/cluster-api-provider-aws/cluster/<cluster-name>
+    sigs.k8s.io/cluster-api-provider-aws/role
+
+These two tags should be present on the following AWS resources:
+
+* VPC
+* All subnets that Cluster API will utilize (private and public)
+* Route tables
+* Internet gateway
+* NAT gateways
+* Elastic IPs assigned to NAT gateways
+
+For the `sigs.k8s.io/cluster-api-provider-aws/cluster/<cluster-name>` tag, a value of "owned" is used by Cluster API itself to indicate resources are owned by a specific cluster, and that the lifecycle of the resource is tied to the lifecycle of the cluster. Cluster API recognizes a value of "shared" to indicate a resource may be shared between multiple clusters, and the lifecycle of said resource is not tied to any particular cluster. In this situation---where AWS infrastructure is _not_ being managed by Cluster API---the value should be something **other** than "owned". Using "shared" is acceptable.
+
+For the `sigs.k8s.io/cluster-api-provider-aws/role` tag, Cluster API uses a variety of values to indicate resources are associated with a particular role. Here are the tag values that should be used:
+
+* VPC, route tables, Internet gateway, and NAT gateways should use the value "common".
+* Public subnets should use the value "public".
+* Private subnets should use the value "private".
+* Elastic IPs (for use by the NAT gateways) should use the value "apiserver".
+
+Additionally, for proper operation of the AWS cloud provider, all subnets where Kubernetes nodes reside should also have the `kubernetes.io/cluster/<cluster-name>` tag present.
```

Comment:
Yes, and additionally private subnets should be tagged with `kubernetes.io/role/internal-elb=1` and public with `kubernetes.io/role/elb=1`, else the cloud provider can't find any subnets to create load balancers in.

Reply:
Added information about these tags via 38ce860.